src/server/active_tran_server.cpp

```diff
@@ -60,6 +60,68 @@ active_tran_server::get_oldest_active_mvccid_from_page_server () const
   return oldest_mvccid;
 }
 
+log_lsa
+active_tran_server::compute_consensus_lsa () const
+{
+  const int total_node_cnt = m_connection_list.size ();
+  const int quorum = total_node_cnt / 2 + 1; // For now, it's fixed to the number of the majority.
+  int cur_node_cnt;
+  std::vector<log_lsa> collected_saved_lsa;
+
+  // TODO The next block has to be exclusive with connection and disconnection
+  {
+    cur_node_cnt = m_page_server_conn_vec.size ();
+    if (cur_node_cnt < quorum)
+      {
+	if (prm_get_bool_value (PRM_ID_ER_LOG_COMMIT_CONFIRM))
+	  {
+	    _er_log_debug (ARG_FILE_LINE,
+			   "compute_consensus_lsa - Quorum unsatisfied: total node count = %d, curreunt node count = %d, quorum = %d\n",
+			   total_node_cnt, cur_node_cnt, quorum);
+	  }
+	return NULL_LSA;
+      }
+    for (const auto &conn : m_page_server_conn_vec)
+      {
+	collected_saved_lsa.emplace_back (conn->get_saved_lsa ());
+      }
+  }
+  /*
+   * Gather all PS'es saved_lsa and sort it in ascending order.
+   * The <cur_node_count - quorum>'th element is the consensus LSA, upon which the majority (quorumn) of PS agrees.
+   * total: 5, cur: 5 - [5, 5, 6, 9, 10] -> "6" is the consensus LSA.
+   * total: 2, cur: 2 - [9, 10] -> "9"
+   * total: 5, cur: 4 - [5, 6, 9, 10] -> "6"
+   * total: 3, cur: 2 - [9, 10] -> "9"
+   */
+  std::sort (collected_saved_lsa.begin (), collected_saved_lsa.end ());
+
+  const auto consensus_lsa = collected_saved_lsa[cur_node_cnt - quorum];
+
+  if (prm_get_bool_value (PRM_ID_ER_LOG_COMMIT_CONFIRM))
+    {
+      constexpr int BUF_SIZE = 1024;
+      char msg_buf[BUF_SIZE];
+      int n = 0;
+      // cppcheck-suppress [wrongPrintfScanfArgNum]
+      n = snprintf (msg_buf, BUF_SIZE,
+		    "compute_consensus_lsa - total node count = %d, current node count = %d, quorum = %d, consensus LSA = %lld|%d\n",
+		    total_node_cnt, cur_node_cnt, quorum, LSA_AS_ARGS (&consensus_lsa));
+      n += snprintf (msg_buf + n, BUF_SIZE - n, "Collected saved lsa list = [ ");
+      for (const auto &lsa : collected_saved_lsa)
+	{
+	  // cppcheck-suppress [wrongPrintfScanfArgNum]
+	  n += snprintf (msg_buf + n, BUF_SIZE - n, "%lld|%d ", LSA_AS_ARGS (&lsa));
+	}
+      snprintf (msg_buf + n, BUF_SIZE - n, "]\n");
+      assert (n < BUF_SIZE);
```

Comment:
```suggestion
	  n += snprintf (msg_buf + n, BUF_SIZE - n, "%lld|%d, ", LSA_AS_ARGS (&lsa));
	}
      snprintf (msg_buf + n - 2, BUF_SIZE - n, " ]\n");
      assert (n < BUF_SIZE);
```
Nit.
How about putting a comma between LSA values? 

it may looks like this :
```
compute_consensus_lsa - total node count = 0, current node count = 0, quorum = 0, consensus LSA = 0|0
Collected saved lsa list = [ 0|1, 1|2, 2|3, 3|4, 4|5 ]
```

Reply:
Hmm.. In my opinion, It doesn't worth making the arguments less readable. Thanks.